src/types/state.js

```diff
@@ -275,13 +275,27 @@ export type UrlState = {|
 
 export type IconState = Set<string>;
 
+// This is used to display the loading progress when fetching a profile from add-on
+export type ProfileLoadingStep =
+  | 'promise'
```

Comment:
These could probably do with a bit more description on what each of the steps mean. It would probably be good to draw a little text diagram explaining the flow of events here.